From d28f52b4452ded9292d9a0816db0de8404cc4d95 Mon Sep 17 00:00:00 2001
From: Wei-Luan Wang <weiluanwang@google.com>
Date: Fri, 24 May 2024 08:27:48 +0000
Subject: [PATCH] Revert "Reland "Make NOTREACHED() without parameters
 [[noreturn]]""

This reverts commit 012bd0ddb09a3105ff68e88895c2abaaab70468f.

Remove the patch after all `NOTREACHED()`` are migrated to
`NOTREACHED_IN_MIGRATION()`.
---
 base/android/feature_map.cc | 4 ++--
 base/check.cc               | 4 ++--
 base/notreached.h           | 6 +++---
 base/version_info/channel.h | 2 +-
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/base/android/feature_map.cc b/base/android/feature_map.cc
index 7ad83122d2..d6d86a8d3c 100644
--- a/base/android/feature_map.cc
+++ b/base/android/feature_map.cc
@@ -38,8 +38,8 @@ const Feature* FeatureMap::FindFeatureExposedToJava(
     return it->second;
   }
 
-  NOTREACHED() << "Queried feature cannot be found in FeatureMap: "
-               << feature_name;
+  NOTREACHED_NORETURN() << "Queried feature cannot be found in FeatureMap: "
+                        << feature_name;
 }
 
 static jboolean JNI_FeatureMap_IsEnabled(JNIEnv* env,
diff --git a/base/check.cc b/base/check.cc
index 6b0a6bf9c0..4f088e3ef6 100644
--- a/base/check.cc
+++ b/base/check.cc
@@ -380,8 +380,8 @@ NotReachedNoreturnError::~NotReachedNoreturnError() {
   log_message_.reset();
 
   // Make sure we die if we haven't.
-  // TODO(crbug.com/40254046): Replace this with NOTREACHED() once LOG(FATAL) is
-  // [[noreturn]].
+  // TODO(crbug.com/40254046): Replace this with NOTREACHED_NORETURN() once
+  // LOG(FATAL) is [[noreturn]].
   base::ImmediateCrash();
 }
 
diff --git a/base/notreached.h b/base/notreached.h
index 9591c6c5ad..3fce471d52 100644
--- a/base/notreached.h
+++ b/base/notreached.h
@@ -48,9 +48,9 @@ namespace logging {
 // milestone is provided the instance is non-fatal (dumps without crashing)
 // until that milestone is hit. That is: `NOTREACHED(base::NotFatalUntil::M120)`
 // starts crashing in M120. See base/check.h.
-#define NOTREACHED(...)                                      \
-  BASE_IF(BASE_IS_EMPTY(__VA_ARGS__), NOTREACHED_NORETURN(), \
-          LOGGING_CHECK_FUNCTION_IMPL(                       \
+#define NOTREACHED(...)                                          \
+  BASE_IF(BASE_IS_EMPTY(__VA_ARGS__), NOTREACHED_IN_MIGRATION(), \
+          LOGGING_CHECK_FUNCTION_IMPL(                           \
               ::logging::NotReachedError::NotReached(__VA_ARGS__), false))
 
 // The DUMP_WILL_BE_NOTREACHED() macro provides a convenient way to
diff --git a/base/version_info/channel.h b/base/version_info/channel.h
index 08c2821844..09cfb09cf2 100644
--- a/base/version_info/channel.h
+++ b/base/version_info/channel.h
@@ -39,7 +39,7 @@ constexpr std::string_view GetChannelString(Channel channel) {
     case Channel::UNKNOWN:
       return "unknown";
   }
-  NOTREACHED();
+  NOTREACHED_NORETURN();
 }
 
 }  // namespace version_info
-- 
2.46.0.rc2.264.g509ed76dc8-goog

